### DISCUSSION TITLE
Native image error

### DISCUSSION THREAD

**Discussion by Angel0926** [quarkusio/quarkus#35524] [Q&A]
hello, I'm doing an upgrade of the quarkus to V3 and my build is running locally, but during the build of the native image I have this error. How to fix it?

Java version: 17.0.8+9-LTS, vendor version: Oracle GraalVM 17.0.8+9.1

[2/8] Performing analysis... [] (52.7s @ 1.32GB)
15,836 (79.45%) of 19,931 types reachable
21,547 (51.47%) of 41,862 fields reachable
78,184 (48.91%) of 159,853 methods reachable
5,309 types, 503 fields, and 5,187 methods registered for reflection
1 native library: -framework *******

Fatal error: com.oracle.graal.pointsto.util.AnalysisError$ParsingError: Error encountered while parsing com.oracle.svm.core.code.FactoryMethodHolder.CsvParser_constructor_55e2e9164aea69da3d88f35a7d7d56fddc9a218d(generated:0)
Parsing context:

Caused by: org.graalvm.compiler.java.BytecodeParser$BytecodeParserError: com.oracle.graal.pointsto.constraints.UnresolvedElementException: Discovered unresolved method during parsing: com.fasterxml.jackson.core.io.IOContext.streamReadConstraints(). This error is reported at image build time because class com.fasterxml.jackson.dataformat.csv.CsvParser is registered for linking at image build time by command line

Caused by: com.oracle.graal.pointsto.constraints.UnresolvedElementException: Discovered unresolved method during parsing: com.fasterxml.jackson.core.io.IOContext.streamReadConstraints(). This error is reported at image build time because class com.fasterxml.jackson.dataformat.csv.CsvParser is registered for linking at image build time by command line

[ERROR] Failed to execute goal io.quarkus.platform:quarkus-maven-plugin:3.0.4.Final:build (default) on project case-version-lambda: Failed to build quarkus application: io.quarkus.builder.BuildException: Build failure: Build failed due to errors

[ERROR] [error]: Build step io.quarkus.deployment.pkg.steps.NativeImageBuildStep#build threw an exception: io.quarkus.deployment.pkg.steps.NativeImageBuildStep$ImageGenerationFailureException: Image generation failed. Exit code: 1

**Comment by gsmet** [comment]
My guess would be that your Jackson versions are not consistent. Could you check that?

**Reply by Angel0926** [reply]
<dependency>
      <groupId>com.fasterxml.jackson.dataformat</groupId>
      <artifactId>jackson-dataformat-csv</artifactId>
      <version>2.15.2</version>
    </dependency>

**Reply by Angel0926** [reply]
What should I do?

**Reply by gsmet** [reply]
which version of Quarkus are you using? The other Jackson components are also in 2.15.2?

If so, please try to come up with a simple reproducer and I will have a look.

**Reply by Angel0926** [reply]
quarkus 3.0.4 final,  <dependency>
            <groupId>io.quarkus</groupId>
            <artifactId>quarkus-resteasy-reactive-jackson</artifactId>
        </dependency>
        <dependency>
            <groupId>io.quarkus</groupId>
            <artifactId>quarkus-rest-client-reactive-jackson</artifactId>
        </dependency>

**Reply by gsmet** [reply]
This is the Jackson you get with 3.0.4: https://github.com/quarkusio/quarkus/blob/3.0.4.Final/bom/application/pom.xml#L95

Your other Jackson dependency should be aligned. I think it's probably defined in our BOM so you should just drop the version and it will be aligned automatically.
